uvdat/core/models/styles.py

```diff
@@ -0,0 +1,290 @@
+            return serialized
+
+        colors = []
+        for color_config in ColorConfig.objects.filter(style=self):
+            color = serialize_fields(color_config, ['name', 'visible', 'single_color'])
+            try:
+                colormap = serialize_fields(
+                    color_config.colormap,
+                    ['discrete', 'clamp', 'n_colors', 'color_by', 'null_color'],
+                )
+                colormap['id'] = color_config.colormap.colormap.id
+                if (
+                    color_config.colormap.range_minimum is not None
+                    and color_config.colormap.range_maximum is not None
+                ):
+                    colormap['range'] = [
+                        float(color_config.colormap.range_minimum),
+                        float(color_config.colormap.range_maximum),
+                    ]
+                color['colormap'] = colormap
+            except ColorConfig.colormap.RelatedObjectDoesNotExist:
+                pass
+            colors.append(color)
+
+        sizes = []
+        for size_config in SizeConfig.objects.filter(style=self):
+            size = serialize_fields(size_config, ['name', 'zoom_scaling', 'single_size'])
+            try:
+                size_range = serialize_fields(
+                    size_config.size_range, ['size_by', 'minimum', 'maximum']
+                )
+                size_range['null_size'] = dict(
+                    size=size_config.size_range.null_size,
+                    transparency=size_config.size_range.null_transparent,
+                )
+                size['size_range'] = size_range
+            except SizeConfig.size_range.RelatedObjectDoesNotExist:
+                pass
+            sizes.append(size)
+
+        filters = []
+        for filter_config in FilterConfig.objects.filter(style=self):
+            filter_ = serialize_fields(filter_config, ['filter_by', 'include', 'transparency'])
+            if filter_config.range_minimum is not None and filter_config.range_maximum is not None:
+                filter_['range'] = [
+                    float(filter_config.range_minimum),
+                    float(filter_config.range_maximum),
+                ]
+            if filter_config.values_list is not None:
+                filter_['list'] = filter_config.values_list
+            filters.append(filter_)
+
+        return dict(
+            default_frame=self.default_frame,
+            opacity=self.opacity,
+            colors=colors,
+            sizes=sizes,
+            filters=filters,
+        )
+
+
+def get_default_colormap():
+    return Colormap.objects.filter(project__isnull=True).first()
+
+
+class ColorConfig(models.Model):
+    style = models.ForeignKey(LayerStyle, related_name='color_configs', on_delete=models.CASCADE)
+    name = models.CharField(max_length=255)
+    visible = models.BooleanField(default=True)
+    single_color = models.CharField(
+        max_length=12, null=True
```

Comment:
```suggestion
        max_length=12, null=True, blank=True
```

As written, in django-admin you cannot set this field to the null value. If you want blank text fields to translate into NULL in the database, always include `blank=True` (at least for anything coming in through django-admin or other standard forms).